binary_port/src/error_code.rs

```diff
@@ -27,27 +27,165 @@ pub enum ErrorCode {
+    #[error("The transaction sent to the network was missing an argument")]
+    InvalidTransactionMissingArg = 45,
+    /// The transaction sent to the network had an argument with an unexpected type
+    #[error("The transaction sent to the network had an unexpected argument type")]
+    InvalidTransactionUnexpectedArgType = 46,
+    /// The transaction sent to the network had an invalid argument
+    #[error("The transaction sent to the network had an invalid argument")]
+    InvalidTransactionInvalidArg = 47,
+    /// The transaction sent to the network had an insufficient transfer amount
+    #[error("The transaction sent to the network had an insufficient transfer amount")]
+    InvalidTransactionInsufficientTransferAmount = 48,
+    /// The transaction sent to the network had a custom entry point when it should have a non
+    /// custom entry point.
+    #[error("The transaction sent to the network should not have a custom entry point")]
+    InvalidTransactionEntryPointCannotBeCustom = 49,
+    /// The transaction sent to the network had a standard entry point when it must be custom.
+    #[error("The transaction sent to the network must have a custom entry point")]
+    InvalidTransactionEntryPointMustBeCustom = 50,
+    /// The transaction sent to the network had empty module bytes
+    #[error("The transaction sent to the network had empty module bytes")]
+    InvalidTransactionEmptyModuleBytes = 51,
+    /// The transaction sent to the network had an invalid gas price conversion
+    #[error("The transaction sent to the network had an invalid gas price conversion")]
+    InvalidTransactionGasPriceConversion = 52,
+    /// The network was unable to calculate the gas limit for the transaction sent.
+    #[error("The network was unable to calculate the gas limit for the transaction sent")]
+    InvalidTransactionUnableToCalculateGasLimit = 53,
+    /// The network was unable to calculate the gas limit for the transaction sent.
+    #[error("The network was unable to calculate the gas limit for the transaction sent.")]
+    InvalidTransactionUnableToCalculateGasCost = 54,
+    /// The transaction sent to the network had an invalid pricing mode
+    #[error("The transaction sent to the network had an invalid pricing mode")]
+    InvalidTransactionPricingMode = 55,
+    /// The transaction sent to the network was invalid for an unspecified reason
+    #[error("The transaction sent to the network was invalid for an unspecified reason")]
+    InvalidTransactionUnspecified = 56,
+    ///As the various enums are tagged non_exhaustive, it is possible that in the future none of
+    /// these previous errors cover the error that occurred, therefore we need some catchall in
+    /// the case that nothing else works.
+    #[error("The transaction or deploy sent to the network was invalid for an unspecified reason")]
+    InvalidTransactionOrDeployUnspecified = 57,
 }
 
 impl TryFrom<u8> for ErrorCode {
```

Comment:
At this stage we might consider to change the error code representation from `u8` to `u16`. We already exhausted 57 out of 255 available error codes. And there will be more coming in the future, when we add, for example `TransactionV2`.

Reply:
This is going to be delivered in https://github.com/casper-network/casper-node/pull/4726